lxd-agent/network.go

```diff
@@ -69,3 +74,59 @@ func serverTLSConfig() (*tls.Config, error) {
 
 	return tlsConfig, nil
 }
+
+func reconfigureNetworkInterfaces() error {
+	// Abort if configuration file does not exist. (Nothing to do).
+	if !shared.PathExists("nics.json") {
+		return nil
+	}
+
+	nicBytes, err := ioutil.ReadFile("nics.json")
+	if err != nil {
+		return err
+	}
+
+	nicData := make(map[string]shared.NicConfig)
+	err = json.Unmarshal(nicBytes, &nicData)
+
+	netDir := "/sys/class/net"
+	files, err := ioutil.ReadDir(netDir)
+	if err != nil {
+		return err
+	}
+
+	for _, f := range files {
+		addrBytes, err := ioutil.ReadFile(filepath.Join(netDir, f.Name(), "address"))
+		if err != nil {
+			return err
+		}
+		macAddress := string(bytes.TrimRight(addrBytes, "\n"))
+
+		nic, ok := nicData[macAddress]
+		if !ok {
+			continue
+		}
+
+		_, err = shared.RunCommand("ip", "link", "set", "down", f.Name())
+		if err != nil {
+			return err
+		}
+
+		_, err = shared.RunCommand("ip", "link", "set", f.Name(), "name", nic.Name)
+		if err != nil {
+			return err
+		}
+
+		_, err = shared.RunCommand("ip", "link", "set", "dev", nic.Name, "mtu", strconv.Itoa(int(nic.Mtu)))
+		if err != nil {
+			return err
+		}
+
+		_, err = shared.RunCommand("ip", "link", "set", "up", nic.Name)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
```

Comment:
@tomponline / @stgraber If this fails it currently stops the agent from running. I wasn't sure if we should just log the errors?

Reply:
agreed